src/utils/APICredential.ts

```diff
@@ -16,10 +16,12 @@ export class APICredentials implements ServiceClientCredentials {
    * {@inheritDoc}
    */
   public signRequest(webResource: WebResource): Promise<WebResource> {
-    if (!webResource.headers) {
-      webResource.headers = {};
-    }
-    webResource.headers["Ocp-Apim-Subscription-Key"] = this.apiKey;
-    return Promise.resolve(webResource);
+    return new Promise(resolve => {
```

Comment:
this is a bad pattern, this function returns a Promise with the updated object that is the original updated object, it's misleading since the caller may think that a copy gets returned.

also I don't see why this is returning a Promise since there's no async operation happening

Reply:
this code implements an interface from the [ms-rest-js](https://github.com/Azure/ms-rest-js) package: https://github.com/Azure/ms-rest-js/blob/master/lib/credentials/serviceClientCredentials.ts so we have to return a Promise like this.

Reply:
ok pls add a note about this